### PR TITLE
HBASE-23295 HBaseContext should use most recent delegation token

### PR DESCRIPTION
**Pull request by iadamcsik-cldr**

### CODE REVIEW

**Comment by petersomogyi** [comment]
retest build

**Comment by petersomogyi** [comment]
retest build

**Comment by petersomogyi** [comment]
retest build

**Comment by petersomogyi** [comment]
retest build

**Comment by iadamcsik-cldr** [comment]
@joshelser @meszibalu Could one of you take a look at this change?

File: spark/hbase-spark-protocol-shaded/pom.xml [hunk 1]
```diff
@@ -87,3 +87,5 @@
   </build>
 
 </project>
+
+
```
Comment (meszibalu) [inline spark/hbase-spark-protocol-shaded/pom.xml]: Please revert these changes.

File: spark/hbase-spark-protocol/pom.xml [hunk 2]
```diff
@@ -74,3 +74,5 @@
   </build>
 
 </project>
+
+
```
Comment (meszibalu) [inline spark/hbase-spark-protocol/pom.xml]: And also these, because they can cause unnecessary conflicts.

File: spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java [hunk 3]
```diff
@@ -70,68 +72,70 @@
   public static final HBaseClassTestRule TIMEOUT =
       HBaseClassTestRule.forClass(TestJavaHBaseContext.class);
 
-  private transient JavaSparkContext jsc;
-  HBaseTestingUtility htu;
-  protected static final Logger LOG = LoggerFactory.getLogger(TestJavaHBaseContext.class);
-
-
+  private static transient JavaSparkContext JSC;
+  private static HBaseTestingUtility HTU;
```
Comment (meszibalu) [inline spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java]: We use to call `HBaseTestingUtility` instances as `TEST_UTIL` in HBase tests.

File: spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java [hunk 4]
```diff
@@ -70,68 +72,70 @@
...
-      htu.deleteTable(TableName.valueOf(tableName));
-      LOG.info("shuting down minicluster");
-      htu.shutdownMiniHBaseCluster();
-      htu.shutdownMiniZKCluster();
-      LOG.info(" - minicluster shut down");
-      htu.cleanupTestDir();
+      HTU.deleteTable(TableName.valueOf(tableName));
     } catch (Exception e) {
-      throw new RuntimeException(e);
+      LOG.info(" - no table " + Bytes.toString(tableName) + " found");
```
Comment (meszibalu) [inline spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java]: Please use parametrized logging:
`LOG.info(" - no table {} found", Bytes.toString(tableName));`

File: spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java [hunk 5]
```diff
@@ -70,68 +72,70 @@
...
     }
-    jsc.stop();
-    jsc = null;
+
+    LOG.info(" - creating table " + Bytes.toString(tableName));
+    HTU.createTable(TableName.valueOf(tableName),
+        new byte[][]{columnFamily, columnFamily1});
+    LOG.info(" - created table");
```
Comment (meszibalu) [inline spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java]: Parametrized logging.

File: spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java (hunk 4 above)
Comment (iadamcsik-cldr) [reply spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java]: These code parts should not have changed, looks like the cherry pick applied more than I asked for, let me revert these parts as well.

Comment (iadamcsik-cldr) [reply spark/hbase-spark/src/test/java/org/apache/hadoop/hbase/spark/TestJavaHBaseContext.java]: Actually, nevermind. I got confused by the weird diff. Will change it.

**Comment by iadamcsik-cldr** [comment]
retest build

**Comment by iadamcsik-cldr** [comment]
Not sure what's going on with the build, the latest build are not available from the jobs' status page and when opening directly they are shown as failed after 0 ms, while the console show the job succeeded.

**Comment by iadamcsik-cldr** [comment]
#51 should be merged first as currently test would fail, but not related to this change. The reason the checks aren't failing is that the unit tests are not being run by yetus, as @petersomogyi noticed.

**Comment by meszibalu** [comment]
Unit tests were run locally.
